stencils/LifecycleViewModel.stencil

```diff
@@ -7,12 +7,19 @@ class LifecycleViewModel<VM: BaseViewModel> {
     let containerView: ContainerView?
 
     init(_ viewModel: VM, containerView: ContainerView? = nil) {
+        assert(Thread.isMainThread, "Constructor must be called on the main thread, but called on \(Thread.current)")
         self.viewModel = viewModel
         self.containerView = containerView
     }
 
     deinit {
-        viewModel.clear()
+        if Thread.current.isMainThread {
```

Comment:
I would probably force clear on the main thread using `DispatchQueue.main.async` so you don't need to check the current thread if it will not break anything of course

Reply:
@dasdranagon from what I understand async always returns immediately, so if this code is called from the main thread (as is the case most of the time) or from another thread it will be scheduled for later (perhaps after other scheduled async calls). This mean even though the Swift ViewModel is "deinited" the kaluga viewmodel inside of it thinks it's alive and well.

Do I misunderstand how `async` works?